### PR TITLE
FIX: Convert raw SQL to TypeORM syntax

### PR DESCRIPTION
**Pull request by r-saba**
This PR removes technical debt introduced in #154 by converting the raw SQL in `event.service.ts` into typeORM syntax.

Having all our queries use the ORM syntax will make our backend easier to maintain and update as new features are added.

I had to update a failing snapshot from develop. It is unrelated to this PR.

*Notes: Discover page only shows public events. This means events with `restriction = 0`

### CODE REVIEW

**Review by ys-lin — COMMENTED** [review]
I ran the tests and there are some test failures...
image

**Review by leojrsilao — APPROVED** [review]
LGTM!! 🚀 🚀  The query looks cleaner now and makes the code more consistent with our other queries.

**Review by cristian-aldea — APPROVED** [review]

**Comment by cristian-aldea** [comment]
Oh one thing I would like to mention is that I think you should revert your changes to the snapshot, since it currently doesn't work for everyone as you outlined in #195, no need to change it.

**Review by cristian-aldea — CHANGES_REQUESTED** [review]
Can you first run snapshots tests on my PR #196 and let me know if it passes. If it does, please revert your snapshot changes. If not I'll reapprove

**Comment by r-saba** [comment]
> Can you first run snapshots tests on my PR #196 and let me know if it passes. If it does, please revert your snapshot changes. If not I'll reapprove

I reverted the snapshot commit.

File: README.md [hunk 1]
```diff
@@ -128,6 +128,12 @@ To test all projects:
 nx run-many --target=test --all
 ```
 
+To test a specific file:
```
Comment (cristian-aldea) [inline README.md:131]: 👍

**Review by cristian-aldea — APPROVED** [review]

**Pull request merged**